File: algorithm/Python3_version/leetcode5429.the-k-strongest-values-in-an-array.py
class Solution: # py3
    def getStrongest(self, arr: List[int], k: int) -> List[int]:
        arr.sort()
        mid = arr[(len(arr)-1)>>1]
        def cmp(x):
            return (abs(x-mid), x)
        arr = sorted(arr,key=cmp,reverse=True)
        return arr[0:k]

File: algorithm/leetcode5429.the-k-strongest-values-in-an-array.cpp
class Solution {
public:
    vector<int> getStrongest(vector<int>& arr, int k) {
        int n = arr.size();
        sort(arr.begin(), arr.end());
        int mid = arr[(n-1)>>1];
        sort(arr.begin(), arr.end(),[&](auto a, auto b){
            return (abs(a-mid) > abs(b-mid)) || (abs(a-mid) == abs(b-mid) && a > b);
        });
        return vector<int> (arr.begin(),arr.begin()+k);
    }
};